src/utils.js

```diff
@@ -18,3 +20,118 @@ export function splitLocation(location) {
+}
+
+export function parseDate(date) {
+  const datePattern = /^(\d{4})-(\d{2})-(\d{2})\s(\d{1,2}):(\d{2}):(\d{2})$/;
+  const [, year, month, day, rawHour, min, sec] = datePattern.exec(date);
+  return new Date(
+    `${year}-${month}-${day}T${("0" + rawHour).slice(-2)}:${min}:${sec}`
+  );
+}
+
+export function addMinutes(date, minutes) {
+  return new Date(date.getTime() + minutes * 60000);
+}
+
+export function compareTime(timeA, timeB) {
+  return (
+    timeA.getHours() === timeB.getHours() &&
+    timeA.getMinutes() === timeB.getMinutes() &&
+    timeA.getSeconds() === timeB.getSeconds()
+  );
+}
+
+export function generateScheduleTime(meetings) {
+  const parsedDates = [];
+  meetings.forEach((meeting) => {
+    parsedDates.push(parseDate(meeting.beginDate));
+  });
+  parsedDates.sort((a, b) => a - b);
+  if (meetings.length === 1) {
+    let day = dayToStr[parsedDates[0].getDay()].toUpperCase();
+    let startTime = parsedDates[0].toLocaleTimeString([], {
+      hour: "2-digit",
+      minute: "2-digit",
+    });
+    let endTime = addMinutes(
+      parsedDates[0],
+      meetings[0].minutesDuration
+    ).toLocaleTimeString([], {
+      hour: "2-digit",
+      minute: "2-digit",
+    });
+    return `${day} ${startTime}-${endTime}`;
+  } else if (meetings.length === 2) {
+    if (compareTime(parsedDates[0], parsedDates[1])) {
+      let firstDay = dayToStr[parsedDates[0].getDay()].toUpperCase();
+      let secondDay = dayToStr[parsedDates[1].getDay()].toUpperCase();
+      let startTime = parsedDates[0].toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+      let endTime = addMinutes(
+        parsedDates[0],
+        meetings[0].minutesDuration
+      ).toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+      return `${firstDay},${secondDay} ${startTime}-${endTime}`;
+    } else {
+      let firstDay = dayToStr[parsedDates[0].getDay()].toUpperCase();
+      let secondDay = dayToStr[parsedDates[1].getDay()].toUpperCase();
+      let firstStartTime = parsedDates[0].toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+      let firstEndTime = addMinutes(
+        parsedDates[0],
+        meetings[0].minutesDuration
+      ).toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+      let secondStartTime = parsedDates[1].toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+      let secondEndTime = addMinutes(
+        parsedDates[1],
+        meetings[1].minutesDuration
+      ).toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+      });
+      return `${firstDay} ${firstStartTime}-${firstEndTime}. ${secondDay} ${secondStartTime}-${secondEndTime}`;
+    }
+  }
+  //TODO: Handle if there are more than 2 different meeting time
+  return "";
+}
+
+export function findSchool(school) {
+  if (missingPrograms[school]) {
```

Comment:
this can be simplified to:

```js
export function findSchool(school) {
  return missingPrograms[school] ?? "";
}
```

Reply:
Lol second time you caught this mistake from me